docs/marionette.collectionview.md

```diff
@@ -742,6 +746,113 @@ The `before:render:collection` event is triggered before the `collectionView`'s
 
 The `render:collection` event is triggered after a `collectionView`'s children have been rendered and buffered. It differs from the `collectionViews`'s `render` event in that it happens __only__ if the `collection` is not not empty.
 
+### "before:render:empty" event
+
+The `"before:render:empty"` event is triggered just after creating a new empty view, but before the view is rendered and added to the DOM.
+
+```js
+var myEmptyView = Marionette.ItemView.extend({
+  template: false
+});
+
+var MyCollectionView = Marionette.CollectionView.extend({
+  emptyView: myEmptyView
+});
+
+var myCollectionView = new MyCollectionView();
+
+myCollectionView.on("before:render:empty", function(){
+  alert("the empty view is now created");
+});
+
+myCollectionView.render()
+```
+
+### "render:empty" event
+
+The `"render:empty"` event is triggered after rendering the empty view and adding it to the view's DOM element.
+
+```js
+var myEmptyView = Marionette.ItemView.extend({
+  template: false
+});
+
+var MyCollectionView = Marionette.CollectionView.extend({
+  emptyView: myEmptyView
+});
+
+var myCollectionView = new MyCollectionView();
+
+myCollectionView.on("render:empty", function(){
+  alert("the empty view has been rendered");
+});
+
+myCollectionView.render()
+```
+
+### "before:remove:empty" event
+
+This is triggered after the empty view instance has been removed, but before it has been destroyed.
```

Comment:
I think its called before it is removed as well. https://github.com/marionettejs/backbone.marionette/blob/minor/src/collection-view.js#L279-L284